parenthetics.py

```diff
@@ -1,28 +1,18 @@
 #!/usr/bin/env python
+from __future__ import unicode_literals
 
 
-def parenthetics(str):
-    parens = []
-    for item in str:
-        parens.append(item)
-    count = 0
-    open_parens = []
-    for item in str:
-        count += 1
-        if item == "(":
-            open_parens.append(count)
-    count_two = 0
-    closed_parens = []
-    for item in str:
-        count_two += 1
-        if item == ")":
-            closed_parens.append(count_two)
-    if len(closed_parens) < len(open_parens):
-        return 1
-    count_final = 0
-    for number in range(len(open_parens) + 1):
-        if open_parens[count_final] > closed_parens[count_final]:
+def parenthetics(str, symbol='()'):
+    open_symbol, closed_symbol = symbol
+    open_count, closed_count = 0, 0
+    for char in str:
+        if char == open_symbol:
+            open_count += 1
+        elif char == closed_symbol:
+            closed_count += 1
+        if open_count < closed_count:
             return -1
-            break
-        count += 1
-    return 0
+    if open_count > closed_count:
+        return 1
+    else:
+        return 0
```

Comment:
This is more work than you need to do.  Keeping the count works, but at the price of the extra variables.  Do you have a data structure you created this week that would help with this?